### PR TITLE
Feat/209/internship projects

### PR DESCRIPTION
**Pull request by CarlosMilano**
# What does this PR do?

This pr add an interns page for users with an applicant role, this means that they log in with LinkedIn, an they can apply to projects that have mentors, also this pr implements the logic to redirect depending on the role that the user have.

# Where should the reviewer start?

app/routes/internshipProjects

# How should this be manually tested?

You need to delete all your cookies, then yo log in with LinkedIn, first you are going to be redirect to /applicationform, after you complete the form you will be redirect to /internshipPorjects, you only have access to some projects, if you click on one of them you will see some description of the project an a Botton that says apply

IMPORTANT: Apply Botton doesn't work yet, @AxelUZ is going to make it work

# Any background context you want to provide?

# What are the relevant tickets?

#209 

# Screenshots

Captura de pantalla 2023-09-18 a la(s) 13 19 40

Captura de pantalla 2023-09-18 a la(s) 13 22 04

Captura de pantalla 2023-09-18 a la(s) 13 22 34

Captura de pantalla 2023-09-18 a la(s) 13 22 55

Captura de pantalla 2023-09-18 a la(s) 13 24 43

Captura de pantalla 2023-09-18 a la(s) 13 25 05

Captura de pantalla 2023-09-18 a la(s) 13 25 29

# Questions

# Checklist

- [ ] I added the necessary documentation, if appropriate.
- [ ] I added tests to prove that my fix is effective or my feature works.
- [ ] I reviewed existing Pull Requests before submitting mine.

### CODE REVIEW

**Review by AxelUZ — COMMENTED** [review]
Everything looks good to me. As Carlos mentions, I will make the "Apply" button work in another ticket.

File: app/core/components/ProposalCard/index.tsx [hunk 1]
```diff
@@ -35,30 +36,43 @@ export const ProposalCard = (props: IProps) => {
...
   const handleCardClick = () => {
-    navigate(`/projects/${props.id}`);
+    if (user) {
+      if (user.role === "ADMIN" || user.role === "USER") {
+        navigate(`/projects/${props.id}`);
+      } else if (user.role === "APPLICANT") {
+        navigate(`/internshipProjects/${props.id}`);
+      }
+    } else {
+      navigate(`/internshipProjects/${props.id}`);
```
Comment (martinrobled0) [inline app/core/components/ProposalCard/index.tsx:49]: Here you are repeating this line, I think that you can avoid that 😛

File: app/core/components/ProposalCard/index.tsx [hunk 2]
```diff
@@ -35,30 +36,43 @@ export const ProposalCard = (props: IProps) => {
...
+    }
   };
 
+  const showPicture = user?.role === "ADMIN" || user?.role === "USER";
```
Comment (martinrobled0) [inline app/core/components/ProposalCard/index.tsx:53]: maybe you can handle this using the authorization.server file

File: app/core/layouts/Header.tsx [hunk 3]
```diff
@@ -27,20 +27,31 @@ const StyledHeaderButton = styled(Button)(({ theme }) => ({
...
+    }
   };
 
   const options: MenuItemArgs[] = [
-    {
-      onClick: handleClickProfile,
-      to: "/",
-      text: "Profile",
-    },
+    ...(currentUser?.role === "ADMIN" || currentUser?.role === "USER"
```
Comment (martinrobled0) [inline app/core/layouts/Header.tsx:46]: also here I think that you can handle with the authorization.server file

File: app/core/layouts/Header.tsx [hunk 4]
```diff
@@ -64,6 +75,19 @@ const Header = ({ title }: IProps) => {
...
 
+  let linkTo = "/internshipProjects";
+
+  if (currentUser) {
+    if (currentUser.role === "ADMIN" || currentUser.role === "USER") {
+      linkTo = "/projects";
+    } else if (currentUser.role === "APPLICANT") {
+      linkTo = "/internshipProjects";
+    }
+  }
```
Comment (martinrobled0) [inline app/core/layouts/Header.tsx:86]: you can improve this code doing something like this:
```
  let linkTo;
    if (currentUser?.role === "APPLICANT") {
      linkTo = "/internshipProjects";
    } else {
      linkTo = "/projects";
    }
```

File: app/core/layouts/Header.tsx [hunk 5]
```diff
@@ -98,24 +122,40 @@ const Header = ({ title }: IProps) => {
...
             <Grid item sx={{ order: { md: 3 } }}>
-              <DropDownButton options={options}>
-                {currentUser?.email}
-              </DropDownButton>
-            </Grid>
-            <Grid item sx={{ marginRight: 2 }}>
-              <Search />
-              &nbsp;
-              <Link to="/projects/create" className="no_decoration">
+              {currentUser ? (
```
Comment (martinrobled0) [inline app/core/layouts/Header.tsx:125]: why this?

File: app/routes/index.tsx [hunk 6]
```diff
@@ -29,6 +29,16 @@ export default function Index() {
 
   const justFour = shuffledInfo.slice(0, 4);
 
+  let navigate = "/internshipProjects";
+
+  if (user) {
```
Comment (martinrobled0) [inline app/routes/index.tsx:34]: I think you can improve this code

File: app/routes/index.tsx [hunk 7]
```diff
@@ -75,17 +85,32 @@ export default function Index() {
...
-            variant="contained"
-            sx={{
-              width: "240px",
-              height: "60px",
-              fontSize: "1.5em",
-            }}
-          >
-            View Projects
-          </Button>
+          {user && (
```
Comment (martinrobled0) [inline app/routes/index.tsx:88]: all types of user has a user, no?

File: app/routes/profile/$email/index.tsx [hunk 8]
```diff
@@ -212,6 +212,8 @@ export const ProfileInfo = () => {
                       variant="outlined"
                       style={{ width: "100%" }}
                       name={"comentario"}
+                      multiline
+                      rows={4}
```
Comment (martinrobled0) [inline app/routes/profile/$email/index.tsx:216]: why this change?

Comment (CarlosMilano) [reply app/routes/profile/$email/index.tsx:216]: visually  looks better

File: app/routes/internshipProjects/index.tsx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+                    }).format(new Date(item.createdAt))}
+                    description={item.description}
+                    status={item.status}
+                    color={item.color}
+                    votesCount={Number(item.votesCount)}
+                    skills={item.searchSkills
+                      .trim()
+                      .split(",")
+                      .map((skill: string) => ({ name: skill }))}
+                    tierName="WZ"
```
Comment (martinrobled0) [inline app/routes/internshipProjects/index.tsx:95]: why "WZ" in those fields?

File: app/routes/internshipProjects/index.tsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+              container
+              spacing={2}
+              sx={{ paddingTop: 2, paddingBottom: 2 }}
+            >
+              {projects.map((item: any, i: number) => (
+                <Grid item xs={12} sm={6} lg={4} key={i}>
+                  <ProposalCard
+                    id={item.id}
+                    title={item.name}
+                    picture="/wizeline.png"
```
Comment (martinrobled0) [inline app/routes/internshipProjects/index.tsx:80]: maybe you could define this as defefault value but in the ProposalCard

File: app/routes/internshipProjects/index.tsx (hunk 9 above)
Comment (martinrobled0) [reply app/routes/internshipProjects/index.tsx:95]: we have a constants file if you really need this

Comment (CarlosMilano) [reply app/routes/internshipProjects/index.tsx:95]: I don't need it

**Review by martinrobled0 — APPROVED** [review]
LGTM!
